.github/workflows/pull_request_and_push.yaml

```diff
@@ -23,3 +23,39 @@ jobs:
         run: pip install -r requirements.lock
       - name: Run pre-commit
         run: pre-commit run --all-files
+
+  build-main-image:
+    name: Build and push main image
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Log in to Red Hat Registry
+      uses: redhat-actions/podman-login@v1
+      with:
+        registry: registry.redhat.io
+        username: ${{ secrets.REGISTRY_REDHAT_IO_USER }}
```

Comment:
Do we have the username configured? seems like the workflow fail due to it missing